azure/functions/decorators/function_app.py

```diff
@@ -1511,6 +1512,57 @@ def decorator():
 
         return wrap
 
+    def mcp_tool_trigger(self,
+                         arg_name: str,
+                         tool_name: str,
+                         description: Optional[str] = None,
+                         tool_properties: Optional[str] = None,
+                         data_type: Optional[Union[DataType, str]] = None,
+                         **kwargs) -> Callable[..., Any]:
+        """
+        The `mcp_tool_trigger` decorator adds :class:`MCPToolTrigger` to the
+        :class:`FunctionBuilder` object for building a :class:`Function` object
+        used in the worker function indexing model.
+
+        This is equivalent to defining `MCPToolTrigger` in the `function.json`,
+        which enables the function to be triggered when MCP tool requests are
+        received by the host.
+
+        All optional fields will be given default values by the function host when
+        they are parsed.
+
+        Ref: https://aka.ms/remote-mcp-functions-python
+
+        :param arg_name: The name of the trigger parameter in the function code.
+        :param tool_name: The logical tool name exposed to the host.
+        :param description: Optional human-readable description of the tool.
+        :param tool_properties: JSON-serialized tool properties/parameters list.
+        :param data_type: Defines how the Functions runtime should treat the
+            parameter value.
+        :param kwargs: Keyword arguments for specifying additional binding
+            fields to include in the binding JSON.
+
+        :return: Decorator function.
+        """
+
+        @self._configure_function_builder
+        def wrap(fb):
+            def decorator():
+                fb.add_trigger(
+                    trigger=MCPToolTrigger(
+                        name=arg_name,
+                        tool_name=tool_name,
+                        description=description,
+                        tool_properties=tool_properties,
```

Comment:
How this toolProperties will support the PropertyeName, PropertyValue, Required parameter?

Reply:
Similar to https://github.com/Azure-Samples/remote-mcp-functions-python/blob/main/src/function_app.py#L14

User defines a ToolProperty and that is assigned to this `tool_properties`